src/v/cluster/partition_probe.cc

```diff
@@ -23,7 +23,16 @@ namespace cluster {
 replicated_partition_probe::replicated_partition_probe(
   const partition& p) noexcept
   : _partition(p)
-  , _public_metrics(ssx::metrics::public_metrics_handle) {}
+  , _public_metrics(ssx::metrics::public_metrics_handle) {
+    config::shard_local_cfg().enable_schema_id_validation.bind().watch(
+      [this]() { reconfigure_metrics(); });
+}
+
+void replicated_partition_probe::reconfigure_metrics() {
+    _metrics.clear();
+    _public_metrics.clear();
+    setup_metrics(_partition.ntp());
+}
```

Comment:
Does this reset all metrics when schema id validation is toggled?  Does everything go back to 0?

Reply:
> Does this reset all metrics when schema id validation is toggled? Does everything go back to 0?

An excellent question I should probably have addressed. 

The values that underlie the metrics have their own lifetime. The reconfiguration only manipulates the "tree" of registration of the view of the metrics. This implementation may not be the most efficient, but it is clear, concise, and relatively easy to understand vs. creating a bunch of state that represents/mirros the actual state of the registration of the metrics. My hope/suspicion is that reconfiguration this isn't particularly common and that that's a reasonable trade-off.

Reply:
> The reconfiguration only manipulates the "tree" of registration of the view of the metrics.

Ah, good point!  Appreciate the clarification!  I'm good with this.